spring-boot-modules/spring-boot-crud/src/main/java/com/baeldung/entity_dto_differences/dto/BookDto.java

```diff
@@ -0,0 +1,35 @@
+package com.baeldung.entity_dto_differences.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.validation.constraints.NotNull;
+
+public class BookDto {
+
+    @NotNull
+    @JsonProperty("NAME")
+    private final String name;
+
+    @NotNull
+    @JsonProperty("AUTHOR")
+    private final String author;
+
+    // Default constructor for Jackson deserialization
```

Comment:
If you use a `@JsonCreator` constructor and move the `@JsonProperty` annotations into that constructor, you can avoid the ugly default constructor. Just an FYI, you don't have to do this if you don't feel like it.